src/services/getZendeskTicket.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  isZendesktTicketResult,
+  ZendeskTicket
+} from '../types/zendeskTicketResult'
+import { base64Encode, makeHttpsRequest } from './httpsRequestUtils'
+import { retrieveZendeskApiSecrets } from './retrieveZendeskApiSecrets'
+import https from 'node:https'
+
+export const getZendeskTicket = async (
+  id: string
+): Promise<ZendeskTicket | undefined> => {
+  const secrets = await retrieveZendeskApiSecrets()
+  const options: https.RequestOptions = {
+    method: 'GET',
+    hostname: secrets.zendeskHostName,
+    path: `/api/v2/tickets/${id}`,
+    headers: {
+      Authorization: base64Encode(
+        `${secrets.zendeskApiUserEmail}/token:${secrets.zendeskApiKey}`
+      )
+    }
+  }
+
+  const data = await makeHttpsRequest(options)
+
+  if (isZendesktTicketResult(data)) {
```

Comment:
What happens if there isn't a ticket at all for the given id? I think the Zendesk api returns a 404 in that case, which I believe at the moment will cause our makeHttpsRequest function throw an error. You might need to tweak that to make this work.

Reply:
I'm currently handling this in the handler....

```
try {
    if (!(await matchZendeskTicket(requestParams))) {
      return await handleUnmatchedRequest(requestParams)
    }
  } catch (error) {
    console.error(error)
    return {
      statusCode: 404,
      body: JSON.stringify({
        message: 'Zendesk ticket not found'
      })
    }
  }

I'm basically assuming that ANY error we get back from the `matchZendeskTicket` function means we stop the process and say the ticket couldn't be found.

Possible errors we could get are errors from the Zendesk API calls (Show Ticket and Show User). One of which would be the legitimate 404 for the Show Ticket call. Do you think we should handle each of these cases individually? Trouble was I couldn't find anything to do with errors in their docs.... just the 200 response.

There is also the possibility that the custom field could not be found, so maybe we also handle that separately.